### PR TITLE
Various optimizations to reduce deadlocks

### PR DESCRIPTION
**Pull request by danielvschoor**
Continuation of #13

### CODE REVIEW

File: aiarena/frontend/templates/match_queue.html [hunk 1]
```diff
@@ -2,8 +2,9 @@
 {% load i18n core_filters %}
 
 {% block content %}
-    {% if requested_matches.count > 0 or round_list.count > 0 %}
-        {% if requested_matches.count > 0 %}
+    {% with requested_count=requested_matches.count %}
+        {% if requested_count > 0 or round_list.count > 0 %}
+        {% if requested_count > 0 %}
```
Comment (lladdy) [inline aiarena/frontend/templates/match_queue.html:7]: is this a duplicate if?

Comment (danielvschoor) [reply aiarena/frontend/templates/match_queue.html:7]: No, it makes sense within this context. I just stored it in a var, so we don't run the same performance intensive query twice

Comment (lladdy) [reply aiarena/frontend/templates/match_queue.html:7]: I meant:
```{% if requested_count > 0 or round_list.count > 0 %}
        {% if requested_count > 0 %}```
But it's not - all good.

**Comment by danielvschoor** [comment]
Merging to test on staging

**Pull request merged**